src/main/java/com/redhat/devtools/intellij/tektoncd/listener/SaveInEditorListener.java

```diff
@@ -0,0 +1,96 @@
+import com.redhat.devtools.intellij.common.utils.JSONHelper;
+import com.redhat.devtools.intellij.common.utils.UIHelper;
+import com.redhat.devtools.intellij.common.utils.YAMLHelper;
+import com.redhat.devtools.intellij.tektoncd.utils.CRDHelper;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_PLURAL;
+import static com.redhat.devtools.intellij.tektoncd.Constants.NOTIFICATION_ID;
+
+public class SaveInEditorListener extends FileDocumentSynchronizationVetoer {
+    Logger logger = LoggerFactory.getLogger(SaveInEditorListener.class);
+
+    @Override
+    public boolean maySaveDocument(@NotNull Document document, boolean isSaveExplicit) {
+        VirtualFile vf = FileDocumentManager.getInstance().getFile(document);
+        // if file is not related to tekton we can skip it
+        if (vf == null || vf.getUserData(KIND_PLURAL).isEmpty()) {
+            return true;
+        }
+
+        String namespace, name, apiVersion;
+        JsonNode spec;
+        CustomResourceDefinitionContext crdContext;
+        try {
+            namespace = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "namespace"});
+            name = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "name"});
+            if (Strings.isNullOrEmpty(namespace) || Strings.isNullOrEmpty(name)) {
+                throw new IOException("Tekton file has not a valid format. Namespace and/or name fields are invalid.");
+            }
+            apiVersion = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"apiVersion"});
+            if (Strings.isNullOrEmpty(apiVersion)) {
+                throw new IOException("Tekton file has not a valid format. ApiVersion field is not found.");
+            }
+            crdContext = CRDHelper.getCRDContext(apiVersion, vf.getUserData(KIND_PLURAL));
+            if (crdContext == null) {
+                throw new IOException("Tekton file has not a valid format. ApiVersion field contains an invalid value.");
+            }
+            spec = YAMLHelper.getValueFromYAML(document.getText(), new String[] {"spec"});
+            if (spec == null) {
+                throw new IOException("Tekton file has not a valid format. Spec field is not found.");
+            }
+        } catch (IOException e) {
+            UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Error"));
+            return false;
+        }
+
+        Notification notification;
+        try (final KubernetesClient client = new DefaultKubernetesClient()) {
```

Comment:
The Kubernetes client should not be created, it is costly and may requires tuning (proxy,...). So you should retrieve it from the root node. Also it would be good from OOP that every interaction with the cluster be implemented through the Tkn interface.